libfreerdp/core/heartbeat.c

```diff
@@ -44,6 +44,40 @@ int rdp_recv_heartbeat_packet(rdpRdp* rdp, wStream* s)
 	         "received Heartbeat PDU -> period=%" PRIu8 ", count1=%" PRIu8 ", count2=%" PRIu8 "",
 	         period, count1, count2);
 
+	if (rdp->heartbeat)
+	{
+		BOOL rc = rdp->heartbeat->ServerHeartbeat(rdp->instance, period, count1, count2);
```

Comment:
Screams for `IFCALL` 
I think most of the time the callback will be `NULL` crashing here? Except in proxy, of course.

Reply:
You're right, thanks